### PR TITLE
Fix Cash payment registry filtering by date or user

### LINKED ISSUE #996: (Bug) Cash payment registry filters result in `400`
The cash payment registry filtering requests cause the server to throw `400 BAD REQUEST` errors. 

On my build this is with the following filters
* Date interval 
* User (the user who created the cash payment)

That leaves reference, cash box and client, all of which seem to work.

### PR DESCRIPTION
**Pull request by mbayopanda**
This PR fix the issue #996

Close #996

### CODE REVIEW

File: server/controllers/finance/cash.js [hunk 1]
```diff
@@ -160,14 +160,20 @@ function listPayment(params) {
...
 
+    if (userId) {
+      params[`u.id`] = userId;
+      delete params.user_id;
+    }
+
     let qc =
       params.dateFrom && params.dateTo ?
-      util.queryCondition(sql, params, null, 'DATE(date) BETWEEN DATE(?) AND DATE(?)') :
+      util.queryCondition(sql, params, null, 'DATE(cash.date) BETWEEN DATE(?) AND DATE(?)') :
```
Comment (jniles) [inline server/controllers/finance/cash.js:176]: :+1:

**Comment by jniles** [comment]
@mbayopanda, could you write one little end-to-end test to verify that the date filters work?  That would be super helpful in preventing this type of bug in the future.

**Comment by mbayopanda** [comment]
no problemo @jniles

**Comment by mbayopanda** [comment]
No problemo @jniles

**Comment by jniles** [comment]
@mbayopanda, I've tested this out locally and it works well.  I'm going to merge this now - please submit your e2e test as a separate PR to ensure that we maintain these filters in the future.

Good work!

**Pull request merged**